### PR TITLE
Fix typecheck errors

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/wp-calypso/pull/78078, p1686135950182729-slack-C02DQP0FP

## Proposed Changes

* Fix the rest errors based on https://github.com/Automattic/wp-calypso/pull/78078

## Testing Instructions

* Run `yarn typecheck` and the errors are gone 🎉

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/data/promote-post/types.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+export enum CampaignStatus {
```
Comment (arthur791004) [inline client/data/promote-post/types.ts:1]: The status is updated via https://github.com/Automattic/wp-calypso/pull/76415/files#diff-3b65ffb0803f6f0a542a7b1373abcae3fa51454ed74898cc3fc6549e98d7ba78, so I guess this is the correct one.

**Review by taipeicoder — APPROVED** [review]
LGTM 👍

File: client/data/promote-post/use-promote-post-campaigns-query.ts [hunk 2]
```diff
@@ -8,58 +9,6 @@ export enum CampaignStatus {
 	TODO2 = 2,
 }
 
-export const AudienceListKeys = {
```
Comment (tyxla) [inline client/data/promote-post/use-promote-post-campaigns-query.ts:11]: Really nice to see this deduplicated 👏

**Review by tyxla — APPROVED** [review]
This is looking good 👍 Nice work! 🚀

**Review by sbarbosa — APPROVED** [review]
LGTM 👍

**Comment by sirbrillig** [comment]
Amazing! I merged #78078 and rebased this over trunk so it can go in as soon as the tests complete. (I guess I could have merged this into the other PR but I didn't consider that until later.)

**Comment by arthur791004** [comment]
Thank you all for the review 🎉

**Pull request merged**